src/sequence_batch_scheduler.cc

```diff
@@ -110,12 +109,116 @@ SequenceBatchScheduler::Create(
   // runner. This is at least 1 even if the model doesn't support
   // batching.
   const size_t model_batch_size = std::max(1, config.max_batch_size());
-  size_t seq_slot_cnt = model_batch_size;
+  sched->seq_slot_cnt_ = model_batch_size;
   if (config.sequence_batching().has_oldest()) {
-    seq_slot_cnt =
+    sched->seq_slot_cnt_ =
         config.sequence_batching().oldest().max_candidate_sequences();
   }
 
+  // Create batchers.
+  RETURN_IF_ERROR(sched->CreateBatchers(new_instances));
+
+
+  SequenceBatchScheduler* raw = sched.release();
+  // Create background threads that watch for different sequence states.
+  raw->StartBackgroundThreads();
+  scheduler->reset(raw);
```

Comment:
Can this be simplified to something like this?
```
  // Create background threads that watch for different sequence states.
  sched->StartBackgroundThreads();
  scheduler->reset(std::move(sched));
```

Reply:
I think we could, but I am not sure about the initial reason for setting the thread on the raw pointer, maybe it has something to do with the `nice` setting.

@tanmayv25 Do you think we can remove the intermediate raw pointer, and set the threads on the unique pointer instead?

Reply:
@kthui  can you elaborate upon the "it has something to do with the nice setting."?

Reply:
I think unique_ptr is only a wrapper around the raw pointer. As per my understanding we can make the proposed simplification.

Reply:
Updated: [Simplify steps before starting background threads](https://github.com/triton-inference-server/core/pull/207/commits/6cdd4eb7b178bcd7803b1506d64668c5de37b855)

Reply:
>  elaborate upon the "it has something to do with the nice setting."?

This is just my example on something that could go wrong, if the original code is trying to avoid some problems by initializing the thread with the raw pointer. The CI is passing, so I think there is no problem.